src/CoreShop2VueStorefrontBundle/Bridge/DocumentMapper/DocumentCategoryMapper.php

```diff
@@ -42,38 +40,40 @@ public function __construct(
 
     /**
      * @param \CoreShop\Component\Product\Model\CategoryInterface $category
-     * @param int $level
-     * @param int $position
-     * @param int $parentId
-     *
-     * @return Category
      */
-    public function mapToDocument(
-        $category,
-        $level = self::CATEGORY_DEFAULT_LEVEL,
-        $position = self::CATEGORY_DEFAULT_POSITION,
-        $parentId = self::CATEGORY_DEFAULT_PARENT_ID
-    ): Category {
+    public function mapToDocument($category): Category
+    {
         $esCategory = $this->documentFactory->getOrCreate(Category::class, $category->getId());
 
         $categoryName = $category->getName() ?: $category->getKey();
+        $parentCategory = $category->getParentCategory();
+
+        $level = $this->documentHelper->countSeparator($category->getFullPath(), '/');
 
         $esCategory->setId($category->getId());
-        $esCategory->setParentId($parentId);
+        $esCategory->setParentId($parentCategory ? $parentCategory->getId() : 0);
         $esCategory->setName($categoryName);
-        $esCategory->setLevel($level);
         $esCategory->setCreatedAt($this->getDateTime($category->getCreationDate()));
         $esCategory->setUpdatedAt($this->getDateTime($category->getModificationDate()));
         $esCategory->setPath($this->documentHelper->buildPath($category));
         $esCategory->setDisplayMode(self::CATEGORY_DEFAULT_DISPLAY_MODE);
         $esCategory->setPageLayout(self::CATEGORY_DEFAULT_PAGE_LAYOUT);
-        $esCategory->setChildrenData($this->buildChildrenData($category->getChildCategories(), ++$level));
+        $esCategory->setChildrenData($this->buildChildrenData($category->getChildCategories(), $level + 1));
         $esCategory->setChildren($this->documentHelper->buildChildrenIds($category->getChildCategories()));
-        $esCategory->setChildrenCount($this->documentHelper->buildChildrenCount($esCategory->children));
+        $esCategory->setChildrenCount($this->documentHelper->countSeparator($esCategory->children, ','));
         $esCategory->setIsAnchor("0");
-        $esCategory->setPosition($position);
-        $esCategory->setUrlKey($this->slugify->slugify($categoryName));
-        $esCategory->setUrlPath($this->documentHelper->buildUrlPath($category->getFullPath()));
+
+        $esCategory->setLevel($level);
+        $esCategory->setPosition($category->getIndex());
+
+        if ($category instanceof \CoreShop2VueStorefrontBundle\Bridge\Model\UrlInterface) {
```

Comment:
Please use import for that, thanks

Reply:
@kwreczycki this one was done the same way it's done a few lines later from before: 

https://github.com/DivanteLtd/coreshop-vsbridge/blob/01e9e07e28f3d454203fee79fd070bfb7558938c/src/CoreShop2VueStorefrontBundle/Bridge/DocumentMapper/DocumentCategoryMapper.php#L77-L85